src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java

```diff
@@ -0,0 +1,115 @@
+package com.exasol.adapter.response.converter;
+
+import com.exasol.adapter.capabilities.*;
+import com.exasol.adapter.json.SchemaMetadataSerializer;
+import com.exasol.adapter.response.CreateVirtualSchemaResponse;
+import com.exasol.adapter.response.DropVirtualSchemaResponse;
+import com.exasol.adapter.response.GetCapabilitiesResponse;
+import com.exasol.adapter.response.PushDownResponse;
+import com.exasol.utils.JsonHelper;
+
+import javax.json.*;
+
+/**
+ * Converts response into JSON format
+ */
+public final class ResponseJsonConverter {
+    private static final String SCALAR_FUNCTION_PREFIX = "FN_";
+    private static final String PREDICATE_PREFIX = "FN_PRED_";
+    private static final String AGGREGATE_FUNCTION_PREFIX = "FN_AGG_";
+    private static final String LITERAL_PREFIX = "LITERAL_";
+    private static final String SCHEMA_METADATA = "schemaMetadata";
+    private static final ResponseJsonConverter responseJsonConverter = new ResponseJsonConverter();
+
+    private ResponseJsonConverter() {
+        //intentionally left blank
+    }
+
+    /**
+     * Returns instance of {@link DropVirtualSchemaResponse} singleton class
+     *
+     * @return {@link DropVirtualSchemaResponse} instance
+     */
+    public static ResponseJsonConverter getInstance() {
+        return responseJsonConverter;
+    }
+
+    /**
+     * Converts drop virtual schema response into a JSON format
+     *
+     * @param dropResponse instance
+     * @return string representation of a JSON Object
+     */
+    @SuppressWarnings("squid:S1172")
+    public String convertDropVirtualSchemaResponse(final DropVirtualSchemaResponse dropResponse) {
+        final JsonBuilderFactory factory = JsonHelper.getBuilderFactory();
```

Comment:
Check whether one factory is enough for all invocations in this class. In that case build it during construction.

Reply:
fixed